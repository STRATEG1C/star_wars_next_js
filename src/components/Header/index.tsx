import React, { useContext } from 'react';
import { AppContext } from '../../context';

const Header: React.FC = () => {
    const { posts, resetPosts } = useContext(AppContext);

    return (
        <header className="header">
            <h2 className="font-bold text-3xl">The Fight Continues</h2>
            <div>
                <span className="font-bold">{ posts.length } Warriors</span>
                <button className="btn btn-blue ml-8 w-32" onClick={resetPosts}>Reset</button>
            </div>
        </header>
    );
};

export default Header;
