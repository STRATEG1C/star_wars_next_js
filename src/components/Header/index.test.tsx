import React from 'react';
import { shallow } from 'enzyme';
import Header from './index';

test('Check render of Header', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
});
