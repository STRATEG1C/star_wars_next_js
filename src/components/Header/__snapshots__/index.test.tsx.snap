// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check render of Header 1`] = `
<header
  className="header"
>
  <h2
    className="font-bold text-3xl"
  >
    The Fight Continues
  </h2>
  <div>
    <span
      className="font-bold"
    >
      0
       Warriors
    </span>
    <button
      className="btn btn-blue ml-8 w-32"
      onClick={[Function]}
    >
      Reset
    </button>
  </div>
</header>
`;
